erpnext/selling/page/sales_funnel/sales_funnel.py

```diff
@@ -1,35 +1,80 @@
 
 from __future__ import unicode_literals
 import frappe
 
 from frappe import _
+from erpnext.accounts.report.utils import convert
+import pandas as pd
 
 @frappe.whitelist()
-def get_funnel_data(from_date, to_date):
+def get_funnel_data(from_date, to_date, company):
 	active_leads = frappe.db.sql("""select count(*) from `tabLead`
 		where (date(`modified`) between %s and %s)
-		and status != "Do Not Contact" """, (from_date, to_date))[0][0]
+		and status != "Do Not Contact" and company=%s""", (from_date, to_date, company))[0][0]
 
 	active_leads += frappe.db.sql("""select count(distinct contact.name) from `tabContact` contact
 		left join `tabDynamic Link` dl on (dl.parent=contact.name) where dl.link_doctype='Customer' 
 		and (date(contact.modified) between %s and %s) and status != "Passive" """, (from_date, to_date))[0][0]
 
 	opportunities = frappe.db.sql("""select count(*) from `tabOpportunity`
 		where (date(`creation`) between %s and %s)
-		and status != "Lost" """, (from_date, to_date))[0][0]
+		and status != "Lost" and company=%s""", (from_date, to_date, company))[0][0]
 
 	quotations = frappe.db.sql("""select count(*) from `tabQuotation`
 		where docstatus = 1 and (date(`creation`) between %s and %s)
-		and status != "Lost" """, (from_date, to_date))[0][0]
+		and status != "Lost" and company=%s""", (from_date, to_date, company))[0][0]
 
 	sales_orders = frappe.db.sql("""select count(*) from `tabSales Order`
-		where docstatus = 1 and (date(`creation`) between %s and %s)""", (from_date, to_date))[0][0]
+		where docstatus = 1 and (date(`creation`) between %s and %s) and company=%s""", (from_date, to_date, company))[0][0]
 
 	return [
 		{ "title": _("Active Leads / Customers"), "value": active_leads, "color": "#B03B46" },
 		{ "title": _("Opportunities"), "value": opportunities, "color": "#F09C00" },
 		{ "title": _("Quotations"), "value": quotations, "color": "#006685" },
 		{ "title": _("Sales Orders"), "value": sales_orders, "color": "#00AD65" }
 	]
+
+@frappe.whitelist()
+def get_opp_by_lead_source(from_date, to_date, company):
+	opportunities = frappe.get_all("Opportunity", filters=[['status', 'in', ['Open', 'Quotation', 'Replied']], ['company', '=', company]], fields=['currency', 'sales_stage', 'opportunity_amount', 'probability', 'source'])
+
+	default_currency = frappe.get_cached_value('Global Defaults', 'None',  'default_currency')
+
+	cp_opportunities = [dict(x, **{'compound_amount': (convert(x['opportunity_amount'], x['currency'], default_currency, to_date) * x['probability']/100)}) for x in opportunities]
+
+	df = pd.DataFrame(cp_opportunities).groupby(['source', 'sales_stage'], as_index=False).agg({'compound_amount': 'sum'})
+
+	result = {}
+	result['labels'] = list(set(df.source.values))
+	result['datasets'] = []
+	
+	for s in set(df.sales_stage.values):
+		result['datasets'].append({'name': s, 'values': [0]*len(result['labels']), 'chartType': 'bar'})
+
+	for i, row in df.iterrows():
```

Comment:
Replace `i` with `_` to show that it will not be used.

Reply:
Thanks for this review @barredterra, I just committed a correction.

Reply:
Actually avoid using `_` as a placeholder, since Frappe already uses that for translations.

Reply:
Humm nice catch! I replaced it with "dummy". Pylint should be ignoring "dummy" according to the regex available in the documentation: Default: _+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_

Reply:
Ah, that's interesting.

I was looking into iteration over a `Dataframe`, and I found this: https://stackoverflow.com/questions/16476924/how-to-iterate-over-rows-in-a-dataframe-in-pandas

I don't know enough about `pandas` or `Dataframe`, but based on that thread, `Dataframe.itertuples` seems to provide better performance. Might be significant when someone's dealing with large datasets.

Just my 2 cents.

Reply:
Thanks for pointing that out @Alchez !
Actually I decided to use Pandas because of it's ability to handle large datasets. It will be very useful for other reports as well.

I have moved to itertuples as suggested by the stackoverflow discussion.
I guess the dataset needs to be quite huge to see any significant difference between the two methods, but it's always better to implement the fastest method from the beginning.

Thanks again!

Reply:
No problem, glad to help! 😄

Reply:
> Actually avoid using `_` as a placeholder, since Frappe already uses that for translations.

Of course, sorry, i didn't think of that.